docs/testplan/pnac-test-plan.md

```diff
@@ -0,0 +1,455 @@
+
+"4"\*            AKA     [2]
+
+"5"\*            SIM     [2]
+
+"6"\*            AKA'    [2]
+
+"7"\*            AKA'    [2]
+
+"8"\*            AKA'    [2]
+
+**Test Case (2.1): Port based NAC (local authentication) on an interface**
+
+**Test Objective:** To verify the Port based NAC (local authentication) on an interface using 
+
+`                              `a valid password. 
+
+**Test Steps:** 1. Log in to the supplicant machine and enter the supplicant file path.
+
+`                       `2. Verify the credential details mentioned in the “wpa\_supplicant.conf” file.
+
+`                       `3. Log in to the switch with preloaded PNAC supported SONiC image.
+
+`                       `4. Verify if the NAC status is UP in global mode and on the specific NAC 
+
+`                            `interface.
+
+`                       `5. Start the supplicant and verify that the authentication is successful.
+
+`           `6. Verify whether the NAC enabled interface is authorized.
+
+`                       `7. Send traffic from supplicant and verify that the traffic is received on the 
+
+`                           `network PC.
+
+**Test Case (2.2): Port based NAC (local authentication) on an interface**
+
+**Test Objective:** To verify the Port based NAC (local authentication) on an interface using 
+
+`                              `an invalid password.
+
+**Test Steps:** 1. Log in to the supplicant machine and enter the supplicant file path.
+
+`                       `2. Verify the credential details mentioned in the “wpa\_supplicant.conf” file.
+
+`                       `3. Log in to the switch with preloaded PNAC supported SONiC image.
+
+`                       `4. Verify if the NAC status is UP in global mode and on the specific NAC
+
+`                           `interface.
+
+`                       `5. Start the supplicant and ensure that the authentication fails.
+
+`           `6. Verify if the NAC enabled interface is unauthorized.
+
+`                       `7. Verify that the network PC does not receive any traffic from the supplicant
+
+`                           `since the authorization is unsuccessful.
+
+**Test Case (2.3): Image Upgrade on the Switch**
+
+**Test Objective:** Verify the local authentication after an image upgradation on the switch.
+
+**Test Steps:** 1. Log in to the switch and upgrade the image which is previously installed.
+
+`                       `2. Verify if the image upgradation is successful.
+
+`                       `3. Verify if the NAC container is accessible.
+
+`                       `4. Start the supplicant.
+
+`                       `5. Verify that the local authentication is successful on the supplicant.
+
+`           `6. Verify the NAC enabled interface is authorized.
+
+`                       `7. Ensure the traffic is received on the network PC after successful 
+
+`                           `authorization.
+
+**Test Case (2.4): Warm Reboot on the Switch**
```

Comment:
Please confirm whether client traffic is being disrupted during Warm Reboot.

Reply:
Yes. there will be traffic disruption until ports are initialized and re-authenticated